matrixcompare uses num_traits::Float without std

This is a bug report.

I tried to use `matrixcompare` as a dependency with a tiny sample `Cargo.toml` shown as the below:
```Cargo.toml
[package]
name = "tmp"
version = "0.1.0"
edition = "2018"

[dependencies]
matrixcompare = "0.1.3"
```
However, `cargo check` failed with the message like:
```console
$ cargo check
    Checking matrixcompare v0.1.3
error[E0432]: unresolved import `num_traits::Float`
 --> /home/kato/.cargo/registry/src/github.com-1ecc6299db9ec823/matrixcompare-0.1.3/src/comparators.rs:5:18
  |
5 | use num_traits::{Float, Num};
  |                  ^^^^^
  |                  |
  |                  no `Float` in the root
  |                  help: a similar name exists in the module (notice the capitalization): `float`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0432`.
error: could not compile `matrixcompare`.

To learn more, run the command again with --verbose.
```

This is caused by that `matrixcompare` uses `num_traits::Float` without the `std` feature of `num-traits`.
I found out from `Cargo.toml` of this crate that it is expected behavior to use `num-traits` without the `std` feature.
https://github.com/Andlon/matrixcompare/blob/8fa1ad31f61abbdd2b40530b2aa99badf14ef642/Cargo.toml#L13

On the other hand, `num-traits` doesn't export [`Float`](
https://github.com/rust-num/num-traits/blob/f6852e047c9dd673078b73400ed666b7d4e1684f/src/lib.rs#L33-L34) without `std` or `libm` features.

So, `matrixcompare` tries to use `num_traits::Float`, which is not exported.

By the way, `matrixcompare` is successfully built on the  CI now. This is due to the [bug](https://github.com/rust-lang/cargo/issues/7916) of `cargo`. 
`matrixcompare` uses `proptest` as a dev-dependency. (and it uses `num-traits` with the default features including `std`!)
This is the reason that `cargo` uses `num-traits` with the `std` feature when building `matrixcompare` as a root crate.
At the last, when I tried to build `matrixcompare` with the nightly version of `cargo`, building failed:
```console
$ cargo +nightly -Z features=dev_dep check
   Compiling autocfg v1.0.1
    Checking matrixcompare-core v0.1.0 (/home/kato/wrk/develop/matrixcompare/matrixcompare-core)
   Compiling num-traits v0.2.12
    Checking matrixcompare v0.1.3 (/home/kato/wrk/develop/matrixcompare)
error[E0432]: unresolved import `num_traits::Float`
 --> src/comparators.rs:5:18
  |
5 | use num_traits::{Float, Num};
  |                  ^^^^^
  |                  |
  |                  no `Float` in the root
  |                  help: a similar name exists in the module (notice the capitalization): `float`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0432`.
error: could not compile `matrixcompare`.

To learn more, run the command again with --verbose.
```

--- a/src/comparators.rs
+++ b/src/comparators.rs
@@ -2,7 +2,7 @@
 
 use crate::ulp::{Ulp, UlpComparisonResult};
 
-use num_traits::{Float, Num};
+use num_traits::{float::FloatCore, Num};
 
 use std::fmt;
 use std::fmt::{Display, Formatter};
@@ -163,7 +163,7 @@ pub struct FloatElementwiseComparator<T> {
 
 impl<T> FloatElementwiseComparator<T>
 where
-    T: Float + Ulp,
+    T: FloatCore + Ulp,
 {
     pub fn default() -> Self {
         let four = T::one() + T::one() + T::one() + T::one();
@@ -192,7 +192,7 @@ where
 
 impl<T> ElementwiseComparator<T> for FloatElementwiseComparator<T>
 where
-    T: Ulp + Float + Display,
+    T: Ulp + FloatCore + Display,
 {
     type Error = UlpError;
 

